storage_service/locations/api/resources.py

```diff
@@ -1685,9 +1687,11 @@ def _attempt_package_request_event(
             )
             request_event.save()
 
-            # Update package status
-            package.status = event_status
-            package.save()
+            # Update package status, if and only if this isn't a deletion
+            # request where we don't want to mistakenly pre-mark it as deleted.
+            if event_status != Package.DELETED:
```

Comment:
It looks like this would always be true — event status would be `Event.DELETE` for deletion.

Reply:
I think I could have commented this better. I still haven't found the right balance for comments/not comments. In this instance, I am trying to be as cautious as possible. Because `_attempt_package_request_event` can handle I think three other event types, I want to make sure this change right now doesn't inadvertently change any of the other status values that might be output by those. So the function here can perform entirely as before when a non `DELETE` request comes in. (Providing I've coded that correctly!)

Reply:
I'll revert this too.